samples/nrf9160/lwm2m_client/src/main.c

```diff
@@ -66,6 +66,31 @@ void client_acknowledge(void)
 	lwm2m_acknowledge(&client);
 }
 
+static void security_handler(enum security_event_type event)
+{
+	switch(event) {
+	case SECURITY_EVT_OFFLINE_REQUEST:
+		LOG_WRN("SECURITY_EVT_OFFLINE_REQUEST");
+		int err = lte_lc_offline();
+
+		if (err) {
+			LOG_ERR("lte_lc_connect() failed %d", err);
```

Comment:
as I suggested earlier, this function should be able to return `int` and then this error code should be returned to caller, so it can stop the lwm2m engine.